### PR TITLE
Cleanup and standardize

### PR DESCRIPTION
**Pull request by jeffcarbs**
- Add rubocop and run autocorrect.
- Clean up gemspec and Gemfile.
  - Move dependencies from Gemfile into gemspec.
  - Remove Gemfile.lock from version control.
- Clean up specs.
  - Use rspec 3.x syntax.
  - Add rspec config file.
- Add example app.
- Update some readme copy.
- Simplify strategy by removing anything already implemented by the `OmniAuth::Strategies::OAuth2` strategy.

### CODE REVIEW

File: omniauth-contactually.gemspec [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require File.expand_path('../lib/omniauth-contactually/version', __FILE__)
+
+Gem::Specification.new do |gem|
+  gem.authors       = 'Contactually'
+  gem.email         = 'api@contactually.com'
+  gem.description   = 'OmniAuth OAuth2 strategy for Contactually.'
+  gem.summary       = gem.description
+  gem.homepage      = 'https://github.com/contactually/omniauth-contactually'
+  gem.licenses      = %w(MIT)
```
Comment (Ortuna) [inline omniauth-contactually.gemspec:9]: We should also create a `LICENSE` file.

File: omniauth-contactually.gemspec [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  gem.files         = `git ls-files`.split("\n")
+  gem.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
+  gem.name          = 'omniauth-contactually'
+  gem.require_paths = %w(lib)
+  gem.version       = OmniAuth::Contactually::VERSION
+
+  # Lock at 1.3.x due to https://github.com/intridea/omniauth-oauth2/issues/81
+  gem.add_dependency 'omniauth-oauth2', '~> 1.3.1'
+
+  gem.add_development_dependency 'simplecov'
```
Comment (Ortuna) [inline omniauth-contactually.gemspec]: Is `simplecov` actually used?

**Comment by Ortuna** [comment]
👍 LGTM apart from optional comments.

File: omniauth-contactually.gemspec (hunk 1 above)
Comment (jeffcarbs) [inline omniauth-contactually.gemspec:9]: Done 👍

**Pull request merged**